src/libraries/Common/src/System/CodeDom/CodeTypeReferenceCollection.cs

```diff
@@ -74,6 +74,15 @@ public void AddRange(CodeTypeReferenceCollection value)
 
         public void Insert(int index, CodeTypeReference value) => List.Insert(index, value);
 
-        public void Remove(CodeTypeReference value) => List.Remove(value);
+        public void Remove(CodeTypeReference value)
```

Comment:
I'm seeing a few issues with this PR:

* The change explicitly throws an exception when the value has not been found, however #39703 proposes that no exception should be thrown at all.
* More importantly, the proposal concerns the explicit interface implementation `IList.Remove(object)` found in the [`CollectionBase` class](https://github.com/dotnet/runtime/blob/144cac3d0ce162444452a5dcc837b904fe149ef4/src/libraries/System.Collections.NonGeneric/src/System/Collections/CollectionBase.cs#L170). This is an unrelated public method defined by a subclass CollectionBase.

Reply:
The CodeTypeReferenceCollection remove method documentation says it is supposed to throw on not found, and the existing code was relaying on the "bug" in collectionbase that was throwing when it wasn't supposed to. The build pipeline/tests was failing after my first fix since it was expecting this exception

This comment will be the same for all my changes to random collection in this pr. I have went over all failing tests and fixes either the test not to expect a throw or fixed the implementation if the documentation said it is supposed to throw on not found

Reply:
Oh I see, thanks for the clarification. I had not realized that the protected `List` property was returning the collection instance itself. The prevalence of this pattern (and accompanying failing tests) suggests to me that the throwing behaviour CollectionBase is part of the informal spec, and changing it might break other users too.